src/core/tree.ts

```diff
@@ -84,6 +85,11 @@ export class Tree<T>
         return this.root.length;
     }
 
+    equals(other: Tree<T>): boolean
+    {
+        return this.root.equals(other.root);
```

Comment:
Is it enough to compare roots with `equals`, that checks if it is the same instance?

Reply:
Ah, ok, this is `equals` from `TreeNode`.